contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,786 @@
+            ? commissionRate - valInfo.commission.rate
+            : valInfo.commission.rate - commissionRate;
+        require(changeRate <= valInfo.commission.maxChangeRate, "INVALID_COMMISSION_RATE");
+
+        valInfo.commission.rate = commissionRate;
+        valInfo.updateTime = block.timestamp;
+
+        emit CommissionRateEdited(operatorAddress, commissionRate);
+    }
+
+    function editDescription(Description calldata description) external whenNotPaused validatorExist(msg.sender) {
+        require(_checkMoniker(description.moniker), "INVALID_MONIKER");
+
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+
+        valInfo.description = description;
+        valInfo.updateTime = block.timestamp;
+
+        emit DescriptionEdited(operatorAddress);
+    }
+
+    function editVoteAddress(
+        bytes calldata newVoteAddress,
+        bytes calldata blsProof
+    ) external whenNotPaused validatorExist(msg.sender) {
+        require(_checkVoteAddress(newVoteAddress, blsProof), "INVALID_VOTE_ADDRESS");
+        require(_voteToOperator[newVoteAddress] == address(0), "DUPLICATE_VOTE_ADDRESS");
+
+        address operatorAddress = msg.sender;
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.updateTime + 1 days <= block.timestamp, "UPDATE_TOO_FREQUENTLY");
+
+        valInfo.voteAddress = newVoteAddress;
+        valInfo.updateTime = block.timestamp;
+        _voteToOperator[newVoteAddress] = operatorAddress;
+
+        emit VoteAddressEdited(operatorAddress, newVoteAddress);
+    }
+
+    function unjail(address operatorAddress) public whenNotPaused validatorExist(operatorAddress) {
+        Validator storage valInfo = _validators[operatorAddress];
+        require(valInfo.jailed, "NOT_JAILED");
+
+        require(
+            IStakeCredit(valInfo.creditContract).getSelfDelegationBNB() >= minSelfDelegationBNB,
+            "NOT_ENOUGH_SELF_DELEGATION"
+        );
+        require(valInfo.jailUntil <= block.timestamp, "STILL_JAILED");
+
+        valInfo.jailed = false;
+        numOfJailed -= 1;
+        emit ValidatorUnjailed(operatorAddress);
+    }
+
+    function delegate(address operatorAddress) external payable whenNotPaused validatorExist(operatorAddress) {
+        uint256 bnbAmount = msg.value;
+        require(bnbAmount >= minDelegationBNBChange, "INVALID_DELEGATION_AMOUNT");
+
+        address delegator = msg.sender;
+        Validator memory valInfo = _validators[operatorAddress];
+
+        if (valInfo.jailed) {
+            // only self delegation
+            require(delegator == operatorAddress, "ONLY_SELF_DELEGATION");
+        }
+
+        uint256 shares = IStakeCredit(valInfo.creditContract).delegate{ value: bnbAmount }(delegator);
+        emit Delegated(operatorAddress, delegator, shares, bnbAmount);
```

Comment:
I think no need to emit the event here, as the `StakeCredit` contract already emit a similar event.

Reply:
Removed